src/doc/trpl/for-loops.md

```diff
@@ -41,3 +41,44 @@ so our loop will print `0` through `9`, not `10`.
 Rust does not have the “C-style” `for` loop on purpose. Manually controlling
 each element of the loop is complicated and error prone, even for experienced C
 developers.
+
+# Enumerate
+
+When you need to keep track of how many times you already looped, you can use the `.enumerate()` function.
+
+## On ranges:
+
+```rust
+for (i,j) in (5..10).enumerate() {
+    println!("i = {} and j = {}", i, j);
+}
+```
+
+Outputs:
+
+```text
+i = 0 and j = 5
+i = 1 and j = 6
+i = 2 and j = 7
+i = 3 and j = 8
+i = 4 and j = 9
+```
+
+Don't forget to add the parentheses around the range.
+
+## On iterators:
+
+```rust
+for (linenumber, line) in lines.enumerate() {
```

Comment:
The integration tests correctly reported that `lines` is undefined. It would be okay if you added something like this:

```rust
# let lines = "hello\nworld".lines();
```

The `#` at the start will be removed before compiling this code but it will tell _rustdoc_ to hide the line from the rendered output.